Sources/LLBBuildSystemUtil/LocalExecutor.swift

```diff
@@ -142,25 +142,13 @@ final public class LLBLocalExecutor: LLBExecutor {
             if exitCode == 0 {
                 uploadFutures = request.outputs.map { output in
                     let outputPath = self.outputBase.appending(RelativePath(output.path))
-                    if output.type == .directory {
-                        return LLBCASFileTree.import(path: outputPath, to: engineContext.db).flatMapError { error in
-                            if case FileSystemError.noEntry = error {
-                                // If we didn't find an output artifact that was a directory, create an empty CASTree to
-                                // represent it.
-                                return LLBCASFileTree.create(files: [], in: engineContext.db).map { $0.id }
-                            }
-                            return engineContext.group.next().makeFailedFuture(error)
-                        }
-                    } else {
-                        var dataIDFuture: LLBFuture<LLBDataID>! = nil
-                        do {
-                            try localFileSystem.readFileContents(outputPath).withData { fileData in
-                                dataIDFuture = client.store(ArraySlice<UInt8>(fileData), type: .plainFile)
-                            }
-                        } catch {
-                            return engineContext.group.next().makeFailedFuture(error)
+                    return LLBCASFileTree.import(path: outputPath, to: engineContext.db).flatMapError { error in
+                        if case FileSystemError.noEntry = error {
```

Comment:
we need to check here that the missing output was a directory, can you add the `output.type == .directory, ` before the `case FileSystemError.noEntry...` check?

Reply:
otherwise we're creating empty directories for missing file outputs